app/models/katello/docker_meta_tag.rb

```diff
@@ -131,6 +131,7 @@ def self.with_identifiers(ids)
 
     def self.cleanup_tags
       self.where(:schema2_id => nil, :schema1_id => nil).delete_all
+      self.where("id not in (?) OR (schema2_id IS NULL AND schema1_id IS NULL)", Katello::RepositoryDockerMetaTag.pluck(:docker_meta_tag_id)).delete_all
```

Comment:
Since we need to do this cleanup against RepositoryDockerMetaTag doesn't it indicate that we don't have our associations configured properly?

Reply:
We don't have dependent deletes on content units from the Repository content unit join models currently. Mostly because they are many to many and also we don't delete content units in katello unless the unit gets deleted in pulp. That deletion happens as part of orphan cleanup task.